concent_api/core/views.py

```diff
@@ -1241,6 +1209,604 @@ def store_message_and_message_status(
         receive_message_status.save()
 
 
+def store_subtask(
+    task_id:                        str,
+    subtask_id:                     str,
+    provider_public_key:            bytes,
+    requestor_public_key:           bytes,
+    state:                          Enum,
+    next_deadline:                  int,
+    task_to_compute:                message.TaskToCompute                = None,
+    report_computed_task:           message.ReportComputedTask           = None,
+    force_get_task_result:          message.concents.ForceGetTaskResult  = None,
+    ack_report_computed_task:       message.AckReportComputedTask        = None,
+    reject_report_computed_task:    message.RejectReportComputedTask     = None,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted = None,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected = None,
+):
+    assert isinstance(task_id,              str)
+    assert isinstance(subtask_id,           str)
+    assert isinstance(provider_public_key,  bytes)
+    assert isinstance(requestor_public_key, bytes)
+    assert state in Subtask.SubtaskState
+
+    if state in Subtask.SUBTASK_ACTIVE_STATES:
+        assert next_deadline is not None
+    elif state in Subtask.SUBTASK_PASSIVE_STATES:
+        assert next_deadline is None
+
+    provider  = Client.objects.get_or_create_full_clean(provider_public_key)
+    requestor = Client.objects.get_or_create_full_clean(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = datetime.datetime.fromtimestamp(next_deadline, timezone.utc),
+    )
+
+    add_messages_to_subtask(
+        subtask,
+        task_to_compute,
+        report_computed_task,
+        force_get_task_result,
+        ack_report_computed_task,
+        reject_report_computed_task,
+        subtask_results_accepted,
+        subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+    return subtask
+
+
+def update_file_status(
+    client_public_key: bytes,
+):
+    force_get_task_result_list = PendingResponse.objects.filter(
+        Q(subtask__requestor__public_key = b64encode(client_public_key)) | Q(subtask__provider__public_key = b64encode(client_public_key)),
+        subtask__state                   = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
+    )
+
+    for pending_reponse in force_get_task_result_list:
+        report_computed_task    = deserialize_message(pending_reponse.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        if get_file_status(file_transfer_token):
+            subtask         = pending_reponse.subtask
```

Comment:
You don't need to create a variable from it, it can be used directly:
```
pending_reponse.subtask = Subtask.SubtaskState.RESULT_UPLOADED.name
...
```